### PR TITLE
fix #483

### LINKED ISSUE #483: #354 fix causes IndexError: string index out of range
## Classification:

Bug

## Reproducibility:

Just run this command through autokey.

``` python
system.exec_command("fcitx-remote -s **fcitx-keyboard-us")
```

## Version

AutoKey version: 0.95.10

Used GUI (Gtk, Qt, or both): 

If the problem is known to be present in more than one version, please list all of those.

Installed via: (PPA, pip3, …). PPA

Linux Distribution: Linux Mint 20

## Summary

#354 adds some commands related to this problem. It causes the result.

## Expected Results

Nothing happens.

## Actual Results

``` python
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 207, in __tryReleaseLock
    self.configManager.lock.release()
RuntimeError: release unlocked lock
2020-11-29 23:28:13,811 ERROR - service - Script error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 485, in execute
    exec(script.code, scope)
  File "<string>", line 2, in <module>
  File "/usr/lib/python3/dist-packages/autokey/scripting.py", line 497, in exec_command
    if output[-1] == "\n":
IndexError: string index out of range
```

### PR DESCRIPTION
**Pull request by fx-kirin**
https://github.com/autokey/autokey/issues/483

Reopening https://github.com/autokey/autokey/pull/484

### CODE REVIEW

File: lib/autokey/scripting.py [hunk 1]
```diff
@@ -494,11 +494,12 @@ def exec_command(self, command, getOutput=True):
                     stdout=subprocess.PIPE,
                     universal_newlines=True) as p:
                 output = p.communicate()[0]
-                if output.endswith("\n"):
-                    # Most shell output has a new line at the end, which we
-                    # don't want. Drop the trailing newline character,
-                    # if the command output something that ends on "\n"
-                    output = output[:-1]
+                if len(output) > 0:
```
Comment (sebastiansam55) [inline lib/autokey/scripting.py:497]: At first glance I had questions about if an error would still occur if the command called had no output but it should be good, it seems that with the addition of `stdout=subprocess.PIPE,` the "output" of a command with no other output (eg a test script like `#!/bin/sh`) will still be a empty string `''`.

Unrelated but is there any reason why we don't swap from
`output[:-1]`
to
`.rstrip()`?

**Review by sebastiansam55 — CHANGES_REQUESTED** [review]
When you open a PR try to include more information in the description than fix x, even a short description is better than nothing, our only active person with git PR approval is not a python expert so do your best to explain what your PR is changing and how, the more information the better. 

This PR is also still on `master` instead of `develop`, when you go to the PR page at the top there are a few dropdowns, make sure you select the right branch for both your fork and for the main autokey fork. I also try to name forks prefixed with what branch it is based on so it is more clear (my fork in the screen shot is `develop_altgr` instead of `altgr`, so it is clear to the approver that the PR is for the develop branch based on the develop branch).

Screenshot from 2020-11-30 09-59-49
Note the 2nd dropdown from the left is on develop, by default github sets it to `master` so you have to change it to `develop` everytime.

Sorry to be so nitpicky! 😬😬😬

**Comment by fx-kirin** [comment]
Thank you for your reviewing. l just wanted to report a bug and show a quick fix of it. I don't have time to go further detail of this change including last one https://github.com/autokey/autokey/issues/354  which causes this bug. You can close this issue and PR if you want and I don't know which implementation is better.

File: lib/autokey/scripting.py (hunk 1 above)
Comment (josephj11) [reply lib/autokey/scripting.py:497]: I don't know Python, but rstrip seems more expressive/clear than indexing the last character in the string.

Comment (josephj11) [reply lib/autokey/scripting.py:497]: Did someone finally copy my Problem Reporting Guide from the autokey-python2 branch to the master? I don't see it. That's why the link points to the old repo.

Comment (josephj11) [reply lib/autokey/scripting.py:497]: The version should probably be 0.95.11 or 0.96 - depending on whether someone wants to put out a minor release before 0.96. I don't think the date of a released version should be modified.

**Review by josephj11 — CHANGES_REQUESTED** [review]

**Comment by josephj11** [comment]
See also #462.

**Comment by BlueDrink9** [comment]
Closed by #489

**Pull request closed without merging**